ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date-range.spec.tsx

```diff
@@ -0,0 +1,247 @@
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+Enzyme.configure({ adapter: new Adapter() })
+import { expect } from 'chai'
+
+import { DateRangeField } from './date-range'
+import moment from 'moment'
+const Common = require('../../js/Common')
+
+//@ts-ignore
+import user from '../singletons/user-instance'
+import { ValueTypes } from '../filter-builder/filter.structure'
+
+const UncontrolledDateRangeField = ({
+  startingValue,
+}: {
+  startingValue: ValueTypes['during']
+}) => {
+  const [value, setValue] = React.useState(startingValue)
+  return (
+    <DateRangeField
+      value={value}
+      onChange={(update) => {
+        setValue(update)
+      }}
+    />
+  )
+}
+
+// do not rely on our own transforms for testing, rely on static data!
+const data = {
+  date1: {
+    timezone: 'America/St_Johns',
+    originalISO: '2021-01-15T06:53:54.316Z',
+    originalDate: new Date('2021-01-15T06:53:54.316Z'),
+    shiftedISO: '2021-01-15T10:23:54.316Z',
+    shiftedDate: new Date('2021-01-15T10:23:54.316Z'),
+    userFormatISO: '2021-01-15T03:23:54.316-03:30',
+    userFormat24: '15 Jan 2021 03:23:54.316 -03:30',
+    userFormat12: '15 Jan 2021 03:23:54.316 am -03:30',
+  },
+  date2: {
+    timezone: 'America/St_Johns',
+    userSuppliedInput: '15 Jan 2021 03:24:54.316 -02:30',
+    parsedOutput: '15 Jan 2021 02:24:54.316 -03:30',
+  },
+  date3: {
+    timezone: 'Etc/UTC',
+    maxFuture: moment().add(10, 'years').toISOString(),
+    disallowedFuture: moment().add(11, 'years').toISOString(),
+  },
+  date4: {
+    timezone: 'America/St_Johns',
+    originalISO: '2021-01-14T06:53:54.316Z',
+    originalDate: new Date('2021-01-14T06:53:54.316Z'),
+    shiftedISO: '2021-01-14T10:23:54.316Z',
+    shiftedDate: new Date('2021-01-14T10:23:54.316Z'),
+    userFormatISO: '2021-01-14T03:23:54.316-03:30',
+    userFormat24: '14 Jan 2021 03:23:54.316 -03:30',
+    userFormat12: '14 Jan 2021 03:23:54.316 am -03:30',
+  },
+}
+describe('verify date field works', () => {
+  before(() => {
+    user.get('user').get('preferences').set('timeZone', data.date1.timezone)
+  })
+  after(() => {
+    user.get('user').get('preferences').set('timeZone', 'Etc/UTC')
+  })
+  beforeEach(() => {
+    user
+      .get('user')
+      .get('preferences')
+      .set('dateTimeFormat', Common.getDateTimeFormats()['ISO'])
+  })
+  afterEach(() => {
+    user
+      .get('user')
+      .get('preferences')
+      .set('dateTimeFormat', Common.getDateTimeFormats()['ISO'])
+  })
+  it(`should not allow overlapping dates`, () => {
+    user
+      .get('user')
+      .get('preferences')
+      .set('dateTimeFormat', Common.getDateTimeFormats()['ISO'])
```

Comment:
❓ Is this needed?

I'm making a few assumptions here. I'm guessing that each `it` is a new test and `beforeEach` should run before each it so this should already be set, right?

Reply:
I think we can remove it.  I'll give it a try.